src/xrc/src/lib.rs

```diff
@@ -380,7 +380,7 @@ impl QueriedExchangeRate {
         let mut inverted_rates: Vec<_> = self
             .rates
             .iter()
-            .map(|rate| utils::invert_rate(*rate))
+            .filter_map(|rate| utils::checked_invert_rate(*rate))
```

Comment:
How about adding a small test where a rate is inverted (or divided), showing that the zero rates are filtered out as expected?

Reply:
Meant to add one. 😓 

Test has been added. Found that we should filter out zero rates in the multiplication as well.

Reply:
Why is it needed there as well? There's no division in multiplication (other than dividing by `RATE_UNIT`).
Sure, multiplication will increase the number of zeroes but that's not problematic. In fact, when we remove the zeroes, they are potentially excluded from the variance.
Or am I missing something?

Reply:
Ah, I thought the plan was to remove the zeros completely. I'll revert that.

Reply:
Reverted and updated the test.